lua/rocks/operations/pin.lua

```diff
@@ -24,7 +24,7 @@ pin.pin = function(rock_name)
     nio.run(function()
         helpers.semaphore.with(function()
             local user_config = helpers.parse_rocks_toml()
-            local rocks_key, user_rock = helpers.get_rock_and_key(user_config, rock_name)
+            local rocks_key, user_rock = helpers.get_rock_and_key(user_config --[[@as MutRocksTomlRef]], rock_name)
```

Comment:
suggestion: I believe you can use an [`---@overload` annotation](https://luals.github.io/wiki/annotations/#overload) to avoid having to cast.

another (preferable) option (that I haven't looked into due to lack of time) could be to extract a shared interface.